packages/core/src/components/Select/Select.tsx

```diff
@@ -94,30 +94,46 @@ type Item = {
   value: string | number;
 };
 
+type Selection = string | string[] | number | number[];
+
 export type SelectProps = {
   multiple?: boolean;
   items: Item[];
   label: string;
   placeholder?: string;
-  onChange: (arg: any) => any;
+  selected?: Selection;
+  onChange: (arg: Selection) => void;
   triggerReset?: boolean;
 };
 
-export const SelectComponent = (props: SelectProps) => {
-  const { multiple, items, label, placeholder, onChange } = props;
+export const SelectComponent = ({
+  multiple,
+  items,
+  label,
+  placeholder,
+  selected,
+  onChange,
+  triggerReset,
```

Comment:
If I'm understanding this correctly, the component is currently in a mixed state of controlled and not controlled. It is a bit confusing to me that the `selected` prop gets copied into internal state, and then the `onChange` is only used as a notification mechanism. Would it make sense to have `selected` directly affect the rendering (without being copied to internal state), and `onChange` telling the surrounding component to update itself as it sees fit and updating the `selected` prop accordingly? This would also get rid of the slightly odd looking `triggerReset` prop, which seems like a leakage of abstraction between the layers. Same thing for the `CheckboxTree`.

Reply:
Alternatively - if it's really intended that this component is to be only self-controlling, the `selected` prop should probably be called `initiallySelected`, but it seems to me that it would still be nice to get rid of the `triggerReset` that feels like it doesn't belong here

Reply:
Yeah, both the tree and the select have these problems. To make this work (the query parameter state), they have to be controlled. I tried to change that, but I'm not sure if this PR is the right place to solve this entirely. I would prefer a separate refactoring of both components. Right now they don't feel very polished